### PR TITLE
Add block attribute into Person class.

### PR DESCRIPTION
**Pull request by DannyDakota**
In order to identify an entry's block, this requires the addition of the block attribute. This is done through the introduction of the block class, Block.java.

To accommodate this change, logic classes (AddCommand.java, EditCommand.java) and test case files were edited.

There are no PRs dependent on this as this PR is intended for v1.3.

This PR is not the finalised graphical UI of the block attribute in the PersonCard. This would be addressed in subsequent PRs.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -114,19 +117,19 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
+        Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Block updatedBlock = editPersonDescriptor.getBlock().orElse(personToEdit.getBlock());
+        Faculty updatedFaculty = editPersonDescriptor.getFaculty().orElse(personToEdit.getFaculty());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        MatriculationNumber updatedMatriculationNumber = editPersonDescriptor.getMatriculationNumber()
+                .orElse(personToEdit.getMatriculationNumber());
+        CovidStatus updatedCovidStatus = editPersonDescriptor.getCovidStatus()
+                .orElse(personToEdit.getStatus());
```
Comment (cwq2326) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:129]: I think you overwritten Kevin's changes for the updated fields, which might cause the test case to fail.

`Name updatedName = editChecker(editPersonDescriptor.getName(), personToEdit.getName());`

**Review by cwq2326 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (DannyDakota) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Thanks for noticing! This must have happened during an incorrectly resolved merged conflict 👍

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -92,6 +98,15 @@ public Person toModelType() throws IllegalValueException {
         }
         final Name modelName = new Name(name);
 
+        if (block == null) {
```
Comment (jiale-c) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:101]: Good flow of throwing exceptions! :ok_hand: Maybe you can put this part into a separate method for checking the validity block of the person?

**Review by jiale-c — APPROVED** [review]
LGTM! Block attribute is added to the files utilises block. :star2: Great job!

**Review by cwq2326 — APPROVED** [review]
Overall code looks good, only thing left to do is to update UG to reflect the new changes.

LGTM 😄

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 2 above)
Comment (DannyDakota) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:101]: I think this is a viable option. I'll consider doing this in a subsequent PR as I believe we should focus on fleshing out the features first before we delve into the coding standards! I'll keep a note on this 👍

**Pull request merged**